platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/constants/Style.java

```diff
@@ -17,39 +20,196 @@
+    }
+
+    /**
+     * Get versioned url of Light style.
+     * <p>
+     * <ul>
+     * <li>Current default version is 9.</li>
+     * </ul>
+     * </p>
+     * <p>
+     * More information on the Mapbox styles API can be found on https://www.mapbox.com/api-documentation/#styles
+     * </p>
+     *
+     * @param version the version of the style.
+     * @return uri to load style from
+     */
+    public static String getLightStyleUrl(int version) {
+        return String.format(Locale.US, LIGHT_BASE, version);
+    }
+
+    /**
+     * Get versioned url of Dark style.
+     * <p>
+     * <ul>
+     * <li>Current default version is 9.</li>
+     * </ul>
+     * </p>
+     * <p>
+     * More information on the Mapbox styles API can be found on https://www.mapbox.com/api-documentation/#styles
+     * </p>
+     *
+     * @param version the version of the style.
+     * @return uri to load style from
+     */
+    public static String getDarkStyleUrl(int version) {
+        return String.format(Locale.US, DARK_BASE, version);
+    }
+
+    /**
+     * Get versioned url of Satellite style.
+     * <p>
+     * <ul>
+     * <li>Current version is 9.</li>
+     * </ul>
+     * </p>
+     * <p>
+     * More information on the Mapbox styles API can be found on https://www.mapbox.com/api-documentation/#styles
+     * </p>
+     *
+     * @param version the version of the style.
+     * @return uri to load style from
+     */
+    public static String getSatelliteStyleUrl(int version) {
+        return String.format(Locale.US, SATELLITE_BASE, version);
+    }
+
+    /**
+     * Get versioned url of Satellite streets style.
+     * <p>
+     * <ul>
+     * <li>Current version is 9.</li>
+     * </ul>
+     * </p>
+     * <p>
+     * More information on the Mapbox styles API can be found on https://www.mapbox.com/api-documentation/#styles
+     * </p>
+     *
+     * @param version the version of the style.
+     * @return uri to load style from
+     */
+    public static String getSatelliteStreetsStyleUrl(int version) {
+        return String.format(Locale.US, SATELLITE_STREETS_BASE, version);
+    }
+
+    /**
      * Indicates the parameter accepts one of the values from {@link Style}.
+     *
+     * @deprecated use dedicated versioned methods in {@link Style} instead.
      */
     @StringDef({MAPBOX_STREETS, EMERALD, LIGHT, DARK, SATELLITE, SATELLITE_STREETS})
     @Retention(RetentionPolicy.SOURCE)
+    @Deprecated
     public @interface StyleUrl {
     }
 
     // IMPORTANT: If you change any of these you also need to edit them in strings.xml
 
     /**
      * Mapbox Streets: A complete basemap, perfect for incorporating your own data.
+     *
+     * @deprecated use {@link #getMapboxStreetsUrl(int)} instead.
      */
-    public static final String MAPBOX_STREETS = "mapbox://styles/mapbox/streets-v8";
+    @Deprecated
+    public static final String MAPBOX_STREETS = "mapbox://styles/mapbox/streets-v9";
```

Comment:
Let's just leave the existing constants pointed at `v8` styles. This will avoid the (admittedly small) chance of any breaking change, and encourage the use of the versioned methods.